lib/responses.js

```diff
@@ -408,6 +408,41 @@ function expectJSON(fn, res) {
   }).catch(handleError(res));
 }
 
+/**
+ * Checks the page size requested to make sure it's a Number and within the maximum.
+ * Returns the default or 0 if there's no default or size is not a Number.
+ * @param {Object} req
+ * @returns {number}
+ */
+function checkPageSize(req) {
+  const size = req.query.size || process.env.CLAY_STORAGE_PAGE_SIZE || 0;
+  const num = Math.abs(Number(size));
+  const max = process.env.CLAY_STORAGE_MAX_PAGE || null;
```

Comment:
Right now setting `max` to `0` will result in the function returning a non-integer.

```js
const max = process.env.CLAY_STORAGE_MAX_PAGE || null;
```

^ Here `process.env.CLAY_STORAGE_MAX_PAGE_SIZE` would be equal to `"0"`, not `0`, so `typeof max === 'string'`.

```
if (max) {
```

^ This condition is `true` for `max === "0"`.

```
return num > max ? max : num;
```

^ This works because javascript will sneakily convert `"0"` to `0`.

But then we ultimately return `"0"` instead of `0` (which doesn't match the type declaration in the docstring).

Reply:
Thanks for catching this. Fixed by coercing `CLAY_STORAGE_MAX_PAGE_SIZE` to a Number first.